spec.html

```diff
@@ -36557,27 +36557,138 @@
+          </li>
+        </ul>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-block-runtime-semantics-evaluation">
+        <h1>Changes to Block Runtime Evaluation</h1>
+        <p>During runtime evaluation of <emu-grammar>Block : `{` StatementList `}`</emu-grammar> (<emu-xref href="#sec-block-runtime-semantics-evaluation"></emu-xref>) the following step is performed in place of step 2:</p>
+        <emu-alg>
+          1. Let _blockEnv_ be NewBlockEnvironment(_oldEnv_).
+        </emu-alg>
+        <p>During runtime evaluation of <emu-grammar>SwitchStatement : `switch` `(` Expression `)` CaseBlock</emu-grammar> (<emu-xref href="#sec-switch-statement-runtime-semantics-evaluation"></emu-xref>) the following step is performed in place of step 4:</p>
+        <emu-alg>
+          1. Let _blockEnv_ be NewBlockEnvironment(_oldEnv_).
+        </emu-alg>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-blockdeclarationinstantiation">
+        <h1>Changes to BlockDeclarationInstantiation</h1>
+        <p>During BlockDeclarationInstantiation (<emu-xref href="#sec-blockdeclarationinstantiation"></emu-xref>) the following step is performed in place of step 2:</p>
+        <emu-alg>
+          1. Assert: _envRec_ is a block Environment Record.
+        </emu-alg>
+        <p>The following steps are performed in place of step 4:</p>
+        <emu-alg>
+          1. Let _functionNames_ be an empty List.
+          1. Let _functionsToInitialize_ be an empty List.
+          1. For each _d_ in _declarations_, in reverse list order do
+            1. For each element _dn_ of the BoundNames of _d_ do
+              1. If IsConstantDeclaration of _d_ is *true*, then
+                1. Perform ! _envRec_.CreateImmutableBinding(_dn_, *true*).
+              1. Else,
+                1. Perform ! _envRec_.CreateMutableBinding(_dn_, *false*).
+            1. If _d_ is a |GeneratorDeclaration| production or a |FunctionDeclaration| production, then
+              1. Let _fn_ be the sole element of the BoundNames of _d_.
+              1. If _fn_ is not an element of _functionNames_, then
+                1. NOTE If there are multiple |FunctionDeclaration|s for the same name, the last declaration is used. There are only multiple declarations if _code_ is not contained in strict mode code.
+                1. Insert _fn_ as the first element of _functionNames_.
+                1. Insert _d_ as the first element of _functionsToInitialize_.
+          1. For each element _d_ in _functionsToInitialize_ do
+            1. Let _fn_ be the sole element of the BoundNames of _d_.
+            1. Let _fo_ be the result of performing InstantiateFunctionObject for _d_ with argument _env_.
+            1. If _d_ is a |FunctionDeclaration| production, then
+              1. Let _fibRec_ be the value of _envRec_.[[FunctionsInBlock]].
+              1. Assert: _fibRec_ is a declarative Environment Record.
```

Comment:
I'll actually just remove this assert. It's implied by "Assert: _envRec_ is a block Environment Record."